### ISSUE TITLE
Tests run to completion, but no screenshots are generated.  What am I doing wrong?

### ISSUE THREAD

**Issue by dpettit** [fastlane/fastlane#8153]
### New Issue Checklist

- [y ] Updated fastlane to the latest version
- [ y] I have read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)

### Issue Description
I use fastlane to create my app from the command line, and am able to get a successful build from it (e.g. I can take the resulting build, and run it successfully in Android Studio).  My ultimate goal is to get Screengrab working properly from the command line, but my thought process was that perhaps it would be helpful to get my tests working in Android Studio first (so that I at least know that they work), before making the jump to getting it all to work from the command line.

As I mentioned, the CLI process runs, and attempts to open the emulator (unsuccessfully so far).  However, the resulting project runs smoothly in Android Studio.

I have created a simple instrumentation test to capture a single screenshot of my Android app using Screengrab, and when I run the test via Android Studio, I receive confirmation messages that "All tests Passed" and that "Tests ran to completion."

However, after the test is completed, I can't find any screenshots inside the project root/fastlane folder.  Could you help me determine what I am doing wrong?  I've been working through this for several weeks now, and unfortunately, I am new to Android (I have submitted 60+ iOS apps using Fastlane Snapshot, however, so thank you!).

##### Complete output when running fastlane, including the stack trace and command used
> You can use: `--capture_output` as the last commandline argument to get that collected for you

[
*************************************************************
Begin BT-A-3 script execution...
*************************************************************
Build ScreengrabDebugAndroidTest and ScreengrabDebug using Gradle...

Incremental java compilation is an incubating feature.
:preBuild UP-TO-DATE
:preScreengrabReleaseBuild UP-TO-DATE
:checkScreengrabReleaseManifest
:preScreengrabDebugBuild UP-TO-DATE
:preScreengrabReleaseAndroidTestBuild UP-TO-DATE
:prepareComAndroidSupportTestEspressoEspressoCore222Library
:prepareComAndroidSupportTestEspressoEspressoIdlingResource222Library
:prepareComAndroidSupportTestExposedInstrumentationApiPublish05Library
:prepareComAndroidSupportTestRules05Library
:prepareComAndroidSupportTestRunner05Library
:prepareComAndroidSupportTestUiautomatorUiautomatorV18212Library
:prepareToolsFastlaneScreengrab055Library
:prepareScreengrabReleaseDependencies
:compileScreengrabReleaseAidl
:compileScreengrabReleaseRenderscript
:generateScreengrabReleaseBuildConfig
:generateScreengrabReleaseResValues
:generateScreengrabReleaseResources
:mergeScreengrabReleaseResources
:processScreengrabReleaseManifest
:processScreengrabReleaseResources
:generateScreengrabReleaseSources
:incrementalScreengrabReleaseJavaCompilationSafeguard
:compileScreengrabReleaseJavaWithJavac
:compileScreengrabReleaseJavaWithJavac - is not incremental (e.g. outputs have changed, no previous execution, etc.).
Note: Some input files use or override a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
:prepareScreengrabReleaseAndroidTestDependencies
:compileScreengrabReleaseAndroidTestAidl
:processScreengrabReleaseAndroidTestManifest
:compileScreengrabReleaseAndroidTestRenderscript
:generateScreengrabReleaseAndroidTestBuildConfig
:generateScreengrabReleaseAndroidTestResValues
:generateScreengrabReleaseAndroidTestResources
:mergeScreengrabReleaseAndroidTestResources
:processScreengrabReleaseAndroidTestResources
:generateScreengrabReleaseAndroidTestSources
:incrementalScreengrabReleaseAndroidTestJavaCompilationSafeguard
:compileScreengrabReleaseAndroidTestJavaWithJavac
Note: /Users/tsdev/Documents/TaleSpring/RandD/TS-Auto-Publish/wip/MyDaddy-1.30-ANDROID/src/androidTestScreengrab/java/com/talespring/wrapper/AndroidHTMLWrapperTest.java uses or overrides a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
:compileScreengrabReleaseAndroidTestNdk UP-TO-DATE
:compileScreengrabReleaseAndroidTestSources
:lintVitalScreengrabReleaseAndroidTest
:mergeScreengrabReleaseAndroidTestShaders
:compileScreengrabReleaseAndroidTestShaders
:generateScreengrabReleaseAndroidTestAssets
:mergeScreengrabReleaseAndroidTestAssets
:transformClassesWithDexForScreengrabReleaseAndroidTest
:mergeScreengrabReleaseAndroidTestJniLibFolders
:transformNative_libsWithMergeJniLibsForScreengrabReleaseAndroidTest
:processScreengrabReleaseAndroidTestJavaRes UP-TO-DATE
:transformResourcesWithMergeJavaResForScreengrabReleaseAndroidTest
:validateSigningScreengrabReleaseAndroidTest
:packageScreengrabReleaseAndroidTest
:assembleScreengrabReleaseAndroidTest
:compileScreengrabReleaseNdk UP-TO-DATE
:compileScreengrabReleaseSources
:lintVitalScreengrabRelease
:mergeScreengrabReleaseShaders
:compileScreengrabReleaseShaders
:generateScreengrabReleaseAssets
:mergeScreengrabReleaseAssets
:transformClassesWithDexForScreengrabRelease
:mergeScreengrabReleaseJniLibFolders
:transformNative_libsWithMergeJniLibsForScreengrabRelease
:processScreengrabReleaseJavaRes
:transformResourcesWithMergeJavaResForScreengrabRelease
:validateSigningScreengrabRelease
:packageScreengrabRelease
:assembleScreengrabRelease

BUILD SUCCESSFUL

Total time: 12.972 secs

This build could be faster, please consider using the Gradle Daemon: https://docs.gradle.org/2.14.1/userguide/gradle_daemon.html

Build complete.
Emulator started, waiting for boot
List of devices attached
* daemon not running. starting it now on port 5037 *
* daemon started successfully *
Hax is enabled
Hax ram_size 0x60000000
HAX is working and emulator runs in fast virt mode.

emulator: Listening for console connections on port: 5554
emulator: Serial number of this emulator (for ADB): emulator-5554
emulator: emulator window was out of view and was recentered

...waiting
...waiting
...waiting
...waiting
...waiting
...waiting
...waiting
...waiting
...waiting
...waiting
...waiting
...waiting
...waiting
...waiting
...waiting
...waiting
...waiting
...waiting
...waiting
...waiting
...waiting
...waiting
...waiting
...waiting
...waiting
...waiting
error: no devices/emulators found
]

### Environment
iMac running MacOS Sierra (10.12.3)
Bash scripts are used to trigger Screengrab and Fastlane
Android Studio 2.2.3

Please run `fastlane env` and copy the output below. This will help us help you :+1:
If you used `--capture_output` option please remove this block - as it is already included there.

[
<details><summary>✅ fastlane environment ✅</summary>

### Stack

| Key                 | Value                                       |
| ------------------- | ------------------------------------------- |
| OS                  | 10.12.3                                     |
| Ruby                | 2.3.3                                       |
| Bundler?            | false                                       |
| Git                 | git version 2.10.1 (Apple Git-78)           |
| Installation Source | ~/.rbenv/versions/2.3.3/bin/fastlane        |
| Host                | Mac OS X 10.12.3 (16D32)                    |
| Ruby Lib Dir        | ~/.rbenv/versions/2.3.3/lib                 |
| OpenSSL Version     | OpenSSL 1.0.2j  26 Sep 2016                 |
| Is contained        | false                                       |
| Is homebrew         | false                                       |
| Xcode Path          | /Applications/Xcode.app/Contents/Developer/ |
| Xcode Version       | 8.2.1                                       |

### System Locale

| Variable | Value       |   |
| -------- | ----------- | - |
| LANG     | en_US.UTF-8 | ✅ |
| LC_ALL   |             |   |
| LANGUAGE |             |   |

### fastlane files:

<details><summary>`./fastlane/Fastfile`</summary>

```ruby
# Customise this file, documentation can be found here:
# https://github.com/fastlane/fastlane/tree/master/fastlane/docs
# All available actions: https://github.com/fastlane/fastlane/blob/master/fastlane/docs/Actions.md
# can also be listed using the `fastlane actions` command

# Change the syntax highlighting to Ruby
# All lines starting with a # are ignored when running `fastlane`

# If you want to automatically update fastlane if a new version is available:
# update_fastlane

# This is the minimum version number required.
# Update this, if you use features of a newer version
fastlane_version "1.104.0"

default_platform :android

platform :android do
  before_all do
    # ENV["SLACK_URL"] = "https://hooks.slack.com/services/..."
  end

  desc "Build debug .apk"
  lane :buildDebug do
    gradle(task: "clean assembleDebug assembleAndroidTest")
  end

  desc "Take a screenshot of each page"
  lane :screenshots do
    #adb(command: "kill-server")
    #adb(command: "start-server")

    #Process.fork {'emulator64-x86 -avd Phone_-_Nexus_4_-_API_19 -no-audio -no-window'}
    #sleep 20
    #adb(command: "wait-for-device")

    
	

    #screengrab
  end

  desc "Submit a new Beta Build to Crashlytics Beta"
  lane :beta do
    gradle(task: "assembleRelease")
    crashlytics

    # sh "your_script.sh"
    # You can also use other beta testing services here
  end

  desc "Deploy a new alpha version (in release mode) to the Google Play Store"
  lane :alpha do
    gradle(
	task: "assemble",
	build_type: "Release",
	project_dir: "android/"
    )
    supply(
	track: "alpha",
	apk: "#{lane_context[SharedValues::GRADLE_APK_OUTPUT_PATH]}"
    )
  end

  desc "Deploy a new version to the Google Play Store"
  lane :deploy do
    gradle(task: "assembleRelease")
    supply
  end

  # You can define as many lanes as you want

  after_all do |lane|
    # This block is called, only if the executed lane was successful

    # slack(
    #   message: "Successfully deployed new App Update."
    # )
  end

  error do |lane, exception|
    # slack(
    #   message: exception.message,
    #   success: false
    # )
  end
end


# More information about multiple platforms in fastlane: https://github.com/fastlane/fastlane/blob/master/fastlane/docs/Platforms.md
# All available actions: https://github.com/fastlane/fastlane/blob/master/fastlane/docs/Actions.md

#pimac:TS_Scripts tsdev$ emulator -list-avds
#Phone_-_Nexus_4_-_API_19
#Phone_-_Nexus_4_-_API_22
#Phone_-_Nexus_4_-_API_23
#Phone_-_Nexus_5X_-_API_22
#Phone_-_Nexus_5_-_API_19
#Phone_-_Nexus_5_-_API_22
#Phone_-_Nexus_5_-_API_23
#Phone_-_Nexus_6P_-_API_22
#Phone_-_Nexus_6P_-_API_23
#Phone_-_Nexus_6_-_API_19
#Phone_-_Nexus_6_-_API_22
#Phone_-_Nexus_6_-_API_23
#Phone_-_Nexus_S_-_API_19
#Phone_-_Nexus_S_-_API_22
#Phone_-_Nexus_S_-_API_23
#Tablet_-_Nexus_10_-_API_19
#Tablet_-_Nexus_10_-_API_22
#Tablet_-_Nexus_10_-_API_23
#Tablet_-_Nexus_7_-_API_19
#Tablet_-_Nexus_7_-_API_22
#Tablet_-_Nexus_7_-_API_23
#Tablet_-_Nexus_7_2012_-_API_19
#Tablet_-_Nexus_7_2012_-_API_22
#Tablet_-_Nexus_7_2012_-_API_23
#Tablet_-_Nexus_9_-_API_19
#Tablet_-_Nexus_9_-_API_22
#Tablet_-_Nexus_9_-_API_23
#Tablet_-_WSVGA_7_-_API_19
#Tablet_-_WSVGA_7_-_API_22
#Tablet_-_WSVGA_7_-_API_23
#Tablet_-_WXGA_10_1_-_API_19
#Tablet_-_WXGA_10_1_-_API_22
#Tablet_-_WXGA_10_1_-_API_23```
</details>

<details><summary>`./fastlane/Appfile`</summary>

```ruby
# Path to the json secret file - Follow https://github.com/fastlane/supply#setup to get one
json_key_file "/Users/tsdev/Documents/TaleSpring/RandD/TS-Auto-Publish/Resources/fastlane-android/keys/supply-key-android.json"
package_name "com.talespring.mydaddy.android"
```
</details>

### fastlane gems

| Gem      | Version | Update-Status |
| -------- | ------- | ------------- |
| fastlane | 2.14.2  | ✅ Up-To-Date  |

### Loaded fastlane plugins:

**No plugins Loaded**

<details><summary><b>Loaded gems</b></summary>

| Gem                       | Version      |
| ------------------------- | ------------ |
| did_you_mean              | 1.0.0        |
| slack-notifier            | 1.5.1        |
| i18n                      | 0.7.0        |
| json                      | 1.8.3        |
| thread_safe               | 0.3.5        |
| tzinfo                    | 1.2.2        |
| minitest                  | 5.8.5        |
| activesupport             | 4.2.7.1      |
| CFPropertyList            | 2.3.4        |
| claide                    | 1.0.1        |
| colored                   | 1.2          |
| nanaimo                   | 0.2.3        |
| xcodeproj                 | 1.4.2        |
| rouge                     | 1.11.1       |
| xcpretty                  | 0.2.4        |
| terminal-notifier         | 1.7.1        |
| unicode-display_width     | 1.1.2        |
| terminal-table            | 1.7.3        |
| plist                     | 3.2.0        |
| public_suffix             | 2.0.4        |
| addressable               | 2.5.0        |
| multipart-post            | 2.0.0        |
| word_wrap                 | 1.0.0        |
| babosa                    | 1.0.2        |
| highline                  | 1.7.8        |
| commander                 | 4.4.3        |
| excon                     | 0.54.0       |
| faraday                   | 0.10.1       |
| unf_ext                   | 0.0.7.2      |
| unf                       | 0.1.4        |
| domain_name               | 0.5.20161129 |
| http-cookie               | 1.0.3        |
| faraday-cookie_jar        | 0.0.6        |
| fastimage                 | 2.0.1        |
| gh_inspector              | 1.0.2        |
| uber                      | 0.0.15       |
| representable             | 2.3.0        |
| retriable                 | 2.1.0        |
| mime-types-data           | 3.2016.0521  |
| mime-types                | 3.1          |
| hurley                    | 0.2          |
| little-plugger            | 1.1.4        |
| multi_json                | 1.12.1       |
| logging                   | 2.1.0        |
| jwt                       | 1.5.6        |
| memoist                   | 0.15.0       |
| os                        | 0.9.6        |
| signet                    | 0.7.3        |
| googleauth                | 0.5.1        |
| httpclient                | 2.8.3        |
| google-api-client         | 0.9.20       |
| mini_magick               | 4.5.1        |
| multi_xml                 | 0.6.0        |
| rubyzip                   | 1.2.0        |
| security                  | 0.1.3        |
| xcpretty-travis-formatter | 0.0.4        |
| dotenv                    | 2.1.1        |
| bundler                   | 1.13.7       |
| faraday_middleware        | 0.10.1       |
| io-console                | 0.4.5        |
</details>

*generated on:* **2017-02-07**
</details>
]

**Comment by dpettit** [comment]
I changed things around, letting fastlane do the build, instead of my shell script.  Also, I discovered that my shell script wasn't actually calling "fastlane screengrab", so of course, nothing was hapening.

Turning my attention to the CLI starting the emulator, part of the issue had to do with permissions.  For some reason, my testing AndroidManifest.xml was not being included, so the READ_EXTERNAL_STORAGE and CHANGE_CONFIGURATION  permissions were not being included.  I just added "debug { manifest.srcFile 'src/androidTestScreengrab/java/AndroidManifest.xml' } to the SourceSets section of my gradle build file, and the emulators began to fully boot.

So now, I can get the emulators up on the CLI, and if I open the project in Android Studio, the App, and the instrumentation tests work great!  I think I'm REALLY close at this point.

SO AT THIS POINT:
The issue is that the CLI is giving me an INSTRUMENTATION_STATUS error of "Unable to find instrumentation info for: ComponentInfo, producing an INSTRUMENTATION_STATUS_CODE of -1.

I then see a message in red that states "Make sure you've used Screengrab.screenshot() in your tests and that your expected tests are being run.  No screenshots were detected."

Although the emulator boots up, my app doesn't start, and the tests do not appear to run.  Not sure where to go from here.
